js/tests/unit/dropdown.spec.js

```diff
@@ -1783,4 +1783,125 @@ describe('Dropdown', () => {
 
     triggerDropdown.dispatchEvent(keydown)
   })
+
+  it('should allow `data-bs-toggle="dropdown"` click events to bubble up', () => {
+    fixtureEl.innerHTML = [
+      '<div class="dropdown">',
+      '  <button class="btn dropdown-toggle" data-bs-toggle="dropdown">Dropdown</button>',
+      '  <div class="dropdown-menu">',
+      '    <a class="dropdown-item" href="#">Secondary link</a>',
+      '  </div>',
+      '</div>'
+    ].join('')
+
+    const btnDropdown = fixtureEl.querySelector('[data-bs-toggle="dropdown"]')
+    const clickListener = jasmine.createSpy('clickListener')
+    const delegatedClickListener = jasmine.createSpy('delegatedClickListener')
+
+    btnDropdown.addEventListener('click', clickListener)
+    document.addEventListener('click', delegatedClickListener)
+
+    btnDropdown.click()
+
+    expect(clickListener).toHaveBeenCalled()
+    expect(delegatedClickListener).toHaveBeenCalled()
+  })
+
+  it('should allow `.dropdown-menu form` click events to bubble up', () => {
+    fixtureEl.innerHTML = [
+      '<div class="dropdown">',
+      '  <button class="btn dropdown-toggle" data-bs-toggle="dropdown">Dropdown</button>',
+      '  <div class="dropdown-menu">',
+      '    <form>',
+      '      <input class="form-control" type="text">',
+      '    </form>',
+      '  </div>',
+      '</div>'
+    ].join('')
+
+    const btnDropdown = fixtureEl.querySelector('[data-bs-toggle="dropdown"]')
+    const form = fixtureEl.querySelector('form')
+    const formInput = form.querySelector('input')
+    const formClickListener = jasmine.createSpy('formClickListener')
+
+    form.addEventListener('click', formClickListener)
+
+    btnDropdown.addEventListener('shown.bs.dropdown', () => {
+      formInput.click()
+      form.click()
+    })
+
+    btnDropdown.addEventListener('hidden.bs.dropdown', () => {
+      throw new Error('should not throw hidden.bs.dropdown event')
+    })
+
+    btnDropdown.click()
+
+    expect(formClickListener).toHaveBeenCalledTimes(2)
+  })
+
+  it('should open the dropdown when clicking `data-bs-toggle="dropdown"`', done => {
+    fixtureEl.innerHTML = [
```

Comment:
This one is not needed :thinking:

Reply:
I added this test because I initially was checking the clicked element using `context._element === SelectorEngine.parents(event.target, SELECTOR_DATA_TOGGLE)[0]`. This meant that the menu would open when clicking on a child of the button, but clicking on the button itself would not open the menu. Without this test, all tests would pass. It would make more sense to combine those two tests though.

Reply:
@rohit2sharma95 I updated/improved this test, but let me know if you still want me to remove it.

Reply:
Still I see this is not needed (since there are tests already for this case)  :slightly_smiling_face: 

/CC @XhmikosR

Reply:
Can you point me to that test? In development, I had "successfully" modified this function so that the menu didn't open and all tests still passed. I added this test to ensure something like that doesn't happen in the future.

Reply:
If I add a `setTimeout` to the `shown.bs.dropdown` in the `data-api -> should show and hide a dropdown` test, it fails as expected when checking for the clicked element using `context._element === SelectorEngine.parents(event.target, SELECTOR_DATA_TOGGLE)[0]`. I'm deleting this test and updating the other one.